### PR TITLE
Fix various compiler warnings

### PR DESCRIPTION
**Pull request by pieterhijma**
This PR fixes a range of compiler warnings.

OndselSolver is compiled within FreeCAD with the following settings:

``` bash
cmake \
      -DCMAKE_C_COMPILER=clang \
      -DCMAKE_CXX_COMPILER=clang++ \
      -DCMAKE_CXX_FLAGS="-Werror=absolute-value -Werror=reorder-ctor -Werror=sign-compare -Werror=overloaded-virtual -Werror=c++11-compat-deprecated-writable-strings -Werror=uninitialized -Werror=unused-parameter -Werror=unused-variable -Werror=nonnull -Werror=overlength-strings -Werror=strncat-size -Werror=strncat-size -Wno-deprecated-declarations" \
     etc.
```

### CODE REVIEW

**Comment by pieterhijma** [comment]
I would recommend to use the above settings in the CI. @sliptonic, agreed?

If so, @adrianinsaval, could you update the CI?
@aiksiongkoh, I recommend to use the above settings in your IDE as well.

**Comment by pieterhijma** [comment]
@chennes, please note that I used even stricter warnings (as errors) than you did, and FreeCAD currently compiles on main with these flags.

**Comment by pieterhijma** [comment]
I now notice that there is another PR with the same name. This is a coincidence and feel free to rename the PR.

File: OndselSolver/ConstraintIJ.cpp [hunk 1]
```diff
@@ -5,13 +5,15 @@
...
+
 #include "ConstraintIJ.h"
+
+#include <utility>
 #include "EndFramec.h"
 
 using namespace MbD;
 
-ConstraintIJ::ConstraintIJ(EndFrmsptr frmi, EndFrmsptr frmj) : frmI(frmi), frmJ(frmj), Constraint()
+ConstraintIJ::ConstraintIJ(EndFrmsptr frmi, EndFrmsptr frmj) : frmI(std::move(frmi)), frmJ(std::move(frmj))
```
Comment (PaddleStroke) [inline OndselSolver/ConstraintIJ.cpp:16]: This change I do not know if it can break something.

**Review by PaddleStroke — APPROVED** [review]
The rest looks fine

**Comment by PaddleStroke** [comment]
I will merge it and test it to make sure it is OK. If it is OK then I'll PR to upstream.

**Pull request merged**